optuna/storages/cached_storage.py

```diff
@@ -0,0 +1,328 @@
+import copy
+import threading
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from optuna import distributions
+from optuna.storages import base
+from optuna.storages.rdb.storage import RDBStorage
+from optuna.study import StudyDirection
+from optuna.study import StudySummary
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+class _TrialUpdates:
```

Comment:
[Question] Could you tell me why this method name is plural? IMO, it can be `_TrialUpdate`.

Reply:
I'll replace the name with `_TrialUpdate`.